tests/test_bigbio.py

```diff
@@ -0,0 +1,660 @@
+        """  # noqa
+        datafolder = Path("biomeddatasets/") / self.NAME
+        datascript = datafolder / self.NAME + ".py"
+        self.assertTrue(datafolder.exists(), "Folder not named " + self.NAME)
+
+        self.assertTrue(
+            datascript.exists(),
+            "Script not named " + self.NAME + ".py",
+        )
+
+    def test_schema(self, task: str):
+        """Search supported tasks within a dataset and verify big-bio schema"""
+
+        mapped_task = self._test_task_exists(task)
+
+        if mapped_task == "kb":
+            log.info("NER/NED/RE/Event Extraction/Coref Task")
+
+            schema = kb_features
+
+            # TODO - change this
+            opt_keys = [
+                "entities",
+                "events",
+                "coreferences",
+                "relations",
+            ]
+
+            needed_keys = [
+                key for key in schema.keys() if key not in opt_keys
+            ]
+
+            for split in self.dataset_bigbio.keys():
+                example = self.dataset_bigbio[split][0]
+
+                # Check for mandatory keys
+                mandatory_keys = all(
+                    [
+                        key in example
+                        for key in schema.keys()
+                        if key not in opt_keys
+                    ]
+                )
+                self.assertTrue(
+                    mandatory_keys,
+                    "id/document_id/passages missing from bigbio view",
+                )
+
+                subkeys = []
+
+                # Ensure tasks that require certain fields are present
+                if task == "ner" or task == "re":
+                    subkeys += ["entities", "relations"]
+
+                    for key in subkeys:
+                        for attrs in schema[key]:
+
+                            self.assertTrue(
+                                self._check_subkey(example[key][0], attrs)
+                            )
+                elif task == "ned":
+                    subkeys += ["entities"]
+
+                    for key in subkeys:
+                        for attrs in schema[key]:
+
+                            self.assertTrue(
+                                self._check_subkey(example[key][0], attrs)
+                            )
+                elif task == "coref":
+                    subkeys += ["coreferences"]
+
+                    for key in subkeys:
+                        for attrs in schema[key]:
+
+                            self.assertTrue(
+                                self._check_subkey(example[key][0], attrs)
+                            )
+
+                elif task == "events":
+                    subkeys += ["events"]
+
+                    for key in subkeys:
+                        for attrs in schema[key]:
+
+                            self.assertTrue(
+                                self._check_subkey(example[key][0], attrs)
+                            )
```

Comment:
maybe construct subkeys in the if/elif conditionals but pull these for loops out and perform afterwards?